torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -314,6 +314,331 @@ TensorView* TensorView::rFactor(const std::vector<int>& axes) {
   return producer;
 }
 
+// Create a TensorView before the original tensor. A common use case is to write
+// results into shared memory or local registers before moving to global memory.
+// Analogous to TVM Cache_Write
+TensorView* TensorView::cache_before() {
+  FusionGuard fg(this->fusion());
+
+  Expr* origin_expr = this->fusion()->origin(this);
```

Comment:
no need for `this->`. There are multiple instances of this pattern, please do a search and replace.